### PR TITLE
fix issue #44 and #25

### PR DESCRIPTION
**Pull request by Marsman1996**
These two issues are actually the same one.
They are both because in the `memcpy` function of https://github.com/rockcarry/ffjpeg/blob/0fa4cf8a86d7f23a3e8336343c1895aa634fdc76/src/jfif.c#L708-L711
ffjpeg copies memory buffer longer than the source memory buffer (i.e. the length of STD_HUFTAB_LUMIN_AC is 178, and the length of STD_HUFTAB_LUMIN_DC is 28, but ffjpeg copies 272 from them).

### CODE REVIEW

File: src/jfif.c [hunk 1]
```diff
@@ -705,10 +705,14 @@ void* jfif_encode(BMP *pb)
     }
 
     // init huffman codec
-    memcpy(jfif->phcac[0]->huftab, STD_HUFTAB_LUMIN_AC, MAX_HUFFMAN_CODE_LEN + 256);
-    memcpy(jfif->phcac[1]->huftab, STD_HUFTAB_CHROM_AC, MAX_HUFFMAN_CODE_LEN + 256);
-    memcpy(jfif->phcdc[0]->huftab, STD_HUFTAB_LUMIN_DC, MAX_HUFFMAN_CODE_LEN + 256);
-    memcpy(jfif->phcdc[1]->huftab, STD_HUFTAB_CHROM_DC, MAX_HUFFMAN_CODE_LEN + 256);
+    memset(jfif->phcac[0]->huftab, 0, MAX_HUFFMAN_CODE_LEN + 256);
```
Comment (rockcarry) [inline src/jfif.c:708]: no need to memset the huftab buffer, because the memory is allocate by calloc.

File: src/jfif.c [hunk 2]
```diff
@@ -705,10 +705,14 @@ void* jfif_encode(BMP *pb)
...
     // init huffman codec
-    memcpy(jfif->phcac[0]->huftab, STD_HUFTAB_LUMIN_AC, MAX_HUFFMAN_CODE_LEN + 256);
-    memcpy(jfif->phcac[1]->huftab, STD_HUFTAB_CHROM_AC, MAX_HUFFMAN_CODE_LEN + 256);
-    memcpy(jfif->phcdc[0]->huftab, STD_HUFTAB_LUMIN_DC, MAX_HUFFMAN_CODE_LEN + 256);
-    memcpy(jfif->phcdc[1]->huftab, STD_HUFTAB_CHROM_DC, MAX_HUFFMAN_CODE_LEN + 256);
+    memset(jfif->phcac[0]->huftab, 0, MAX_HUFFMAN_CODE_LEN + 256);
+    memset(jfif->phcac[1]->huftab, 0, MAX_HUFFMAN_CODE_LEN + 256);
+    memset(jfif->phcdc[0]->huftab, 0, MAX_HUFFMAN_CODE_LEN + 256);
+    memset(jfif->phcdc[1]->huftab, 0, MAX_HUFFMAN_CODE_LEN + 256);
+    memcpy(jfif->phcac[0]->huftab, STD_HUFTAB_LUMIN_AC, 178);
```
Comment (rockcarry) [inline src/jfif.c:712]: it's better to use sizeof(xx) instead. but you should also modify the declare of STD_HUFTAB_LUMIN_AC in huffman.h, like:
extern const BYTE STD_HUFTAB_LUMIN_AC[178];

**Review by rockcarry — COMMENTED** [review]
use sizeof

File: src/jfif.c (hunk 1 above)
Comment (Marsman1996) [reply src/jfif.c]: OK, I have deleted these code

**Comment by Marsman1996** [comment]
update code with `sizeof()`

**Review by rockcarry — APPROVED** [review]
look's good.
thank you.

**Pull request merged**